files/usr/lib/jeos-firstboot

```diff
@@ -470,4 +470,30 @@ if [ "$config_wireless" = "true" ]; then
     run ifdown $wlan_device 2>/dev/null || true
     run ifup $wlan_device || true
 fi
+
+if [ -d "/usr/share/jeos-firstboot" ]; then
```

Comment:
This would be nicer:

```
modules=()

if pushd "/usr/share/jeos-firstboot" &>/dev/null; then
        for module in *; do
        source $module && modules+=("$module")
    done
popd &>/dev/null
```

Then the rest doesn't need to be inside the if. I'd prefer it if we had a generic `call_module_hook` function, like this:

```
call_module_hook() {
    hook="$1"
    for module in "${module[@]}"; do
        [ $(type -f "${module}_$hook") == "function" ] || continue
        "${module}_$hook" && true # To not trigger errexit
        [ $? -eq 0 ] || return $?
    done
    return 0
}
```

and could use this in the appropriate places.

Reply:
Should call_module_hook be in functions file?

Reply:
And is this little bit wrong as it should be hook1_function, hook1_post and hook1_cleanup and after that hook2_function, hook2_post and hook2_cleanup. As this is just cosmetic and actually doesn't make any difference.

Reply:
My idea about the `_post` and `_cleanup` was different - it's supposed to run in appropriate places in `jeos-firstboot`, not for each hook.

`jeos-firstboot` would contain

`trap EXIT "call_module_hook cleanup"` for example.

Reply:
Ok i've little bit updated script to fit this idea. I don't know it trap exit is needed as call_module_hook checks if there is cleanup or post function?